text/0000-project-descriptor-images.md

```diff
@@ -0,0 +1,159 @@
+
+Example:
+
+```toml
+[[images]]
+registry = "gcr.io"
+repository = "spring-petclinic"
+tags = ["latest", "v1"]
+
+[[images]]
+repository = "spring-petclinic"
+tags = ["latest", "v1"]
+
+[[images]]
+registry = "private.registry.corp:8443"
+repository = "spring-petclinic"
+```
+
+This example will produce the following images:
+
+- `gcr.io/spring-petclinic:latest`
+- `gcr.io/spring-petclinic:v1`
+- `docker.io/spring-petclinic:latest`
+- `docker.io/spring-petclinic:v1`
+- `private.registry.corp:8443/spring-petclinic:latest`
+
+# Alternatives
+[alternatives]: #alternatives
+
+- have `registries` as an array:
+
+  ```toml
+  [images]
+  registries = ["gcr.io", "docker.io"]
+  repository = "spring-petclinic"
+  tags = ["latest", "v1"]
+  ```
+
+- have `tag` as a string instead of `tags` array:
+
+  ```toml
+  [[images]]
+  registry = "gcr.io"
+  repository = "spring-petclinic"
+  tag = "latest"
+
+  [[images]]
+  registry = "docker.io"
+  repository = "spring-petclinic"
+  tag = "v1"
+
+  # ...
+  ```
+
+- Simple array of strings containing all information i.e. `registry`, `repository`, `tag`
+
+  ```toml
+  images = [
+      "gcr.io/spring-petclinic:latest",
+      "gcr.io/spring-petclinic:v1",
+      "docker.io/spring-petclinic:latest",
+      "docker.io/spring-petclinic:v1"
+  ]
+  ```
+
+- Simple array of images containing all information with `repository` being a placeholder
+
+  ```toml
+  repository = "spring-petclinic"
+  images = [
+      "gcr.io/{repository}:latest",
+      "gcr.io/{repository}:v1",
+      "docker.io/{repository}:latest",
+      "docker.io/{repository}:v1"
+  ]
+  ```
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [RFC 0019 Minimal Project Descriptor](./0019-project-descriptor.md)
+- [Rejected RFC PR#16](https://github.com/buildpacks/rfcs/pull/16)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Naming (`images` / `targets` / other)?
+- Should `project.version` as the default value for `images.tags`?
+
+# Spec. Changes
+[spec-changes]: #spec-changes
+
+The following will be added to the [Project Descriptor Extension Spec](https://github.com/buildpacks/spec/blob/extensions/project-descriptor%2F0.2/extensions/project-descriptor.md):
+
+### `io.buildpacks.images` (optional)
+
+This table MAY contain an array of image repository label definitions. The schema for this table is:
+
+```
+[[io.buildpacks.images]]
+registry = "<string (optional default=docker.io)"
+repository = "<string (optional, default=io.buildpacks.name)>"
+tags = ["<string (optional, default=latest)>"]
+```
+
+- `registry` - a `REGISTRY_HOST[:REGISTRY_PORT]` component of the image name.
+- `repository` - a `REPOSITORY` component of the image name.
+- `tags` - a list of tags used as the `TAG` component of the image name.
+
+If no `registry` is provided, the default value MUST be `"docker.io"`.
+
+If no `repository` component is provided, the default value MUST be derived from the `io.buildpacks.name` field.
```

Comment:
```suggestion
If no `repository` component is provided, the default value MUST be derived from the `io.buildpacks.id` field.
```

I just realized, that `name` is supposed to be human readable and hence likely unfit to serve as image name.